modules/tag-policy-assignment/variables.tf

```diff
@@ -0,0 +1,18 @@
+variable "ou_required_tags" {
+  type = map(object({
+    values       = optional(list(string))
+    enforced_for = optional(list(string))
+  }))
+  description = "Per tag key, a map of tag properties"
+}
+
+variable "tags" {
+  type        = map(string)
+  description = "Map of tags"
```

Comment:
because the module is also about tags it might be good to be explicit that this is about the tags on the resource